fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultScheduledDateGenerator.java

```diff
@@ -96,10 +96,33 @@ public LocalDate generateNextRepaymentDate(final LocalDate lastRepaymentDate, fi
                         loanApplicationTerms.getNumberOfdays());
             }
         }
-        
+        if (currentCalendar == null && holidayDetailDTO.getWorkingDays().getExtendTermForRepaymentsOnHolidays()) {
```

Comment:
Tried calling adjustRepaymentDate. It doesn't reschedule for holidays other than weekend. Went through the code of adjustRepaymentDate and found nothing that handles holidays. The code block after the if does something which I couldn't find in any existing function in the codebase.